android/app/src/main/java/com/coboltforge/dontmind/multivnc/VncCanvasActivity.java

```diff
@@ -779,10 +779,7 @@ public void onClick(DialogInterface dialog, int id) {
 	 * color mode (already done), scaling
 	 */
 	void setModes() {
-
-		AbstractScaling.getByScaleType(connection.getScaleMode())
-				.setScaleTypeForActivity(this);
-
+		new ZoomScaling().setScaleTypeForActivity(this);
```

Comment:
We directly create an object of `ZoomScaling`. It removes the need of static factory methods in `AbstractScaling`.